### PR TITLE
fix bugzilla Issue 24822 - When passing a non-POD argument to an rvalue parame…

### PR DESCRIPTION
**Pull request by WalterBright**
…ter, an unnecessary blit is done

Since this is an optimization, not a decent way to test it.

### CODE REVIEW

**Review by thewilsonator — DISMISSED** [review]
Please retitle the _commit message_: "fix bugzilla Issue 24822 - When passing a non-POD..."

**Review by thewilsonator — COMMENTED** [review]
test cases?

**Comment by WalterBright** [comment]
As I mentioned, no decent way to test it.

File: compiler/src/dmd/e2ir.d [hunk 1]
```diff
@@ -5504,10 +5504,14 @@ elem *callfunc(const ref Loc loc,
                 /* Copy to a temporary, and make the argument a pointer
                  * to that temporary.
                  */
+                auto arge = arg;
+                while (arge.op == EXP.comma)
+                    arge = arge.isCommaExp().e2;
+
```
Comment (dkorpel) [inline compiler/src/dmd/e2ir.d:5510]:
```suggestion
```

File: compiler/src/dmd/e2ir.d [hunk 2]
```diff
@@ -5504,10 +5504,14 @@ elem *callfunc(const ref Loc loc,
...
                 VarDeclaration v;
-                if (VarExp ve = arg.isVarExp())
+                if (VarExp ve = arge.isVarExp())
```
Comment (dkorpel) [inline compiler/src/dmd/e2ir.d:5512]:
```suggestion
                if (VarExp ve = arge.lastComma().isVarExp())
```

**Review by dkorpel — APPROVED** [review]

**Pull request merged**